dissect/target/plugins/os/unix/config.py

```diff
@@ -81,54 +105,216 @@ def parse_file(self, fh: io.TextIO) -> None:
+
+        for field_name in ["collapse", "seperator", "comment_prefixes"]:
+            value = getattr(contents, field_name, None) or getattr(self, field_name)
+            if value:
+                kwargs.update({field_name: value})
+
+        return self.parser(**kwargs)
+
+
+CONFIG_MAP: dict[str, ParserConfig] = {
+    "ini": ParserConfig(Ini),
+    "xml": ParserConfig(Txt),
+    "json": ParserConfig(Txt),
+    "cnf": ParserConfig(Default),
+    "conf": ParserConfig(Default, seperator=(r"\s")),
+    "sample": ParserConfig(Txt),
+    "template": ParserConfig(Txt),
 }
-KNOWN_FILES: dict[str, LinuxConfigurationParser] = {
-    "ulogd.conf": Ini,
+KNOWN_FILES: dict[str, type[ConfigurationParser]] = {
+    "ulogd.conf": ParserConfig(Ini),
+    "hosts.allow": ParserConfig(Default, seperator=(":",), comment_prefixes=("#",)),
+    "hosts.deny": ParserConfig(Default, seperator=(":",), comment_prefixes=("#",)),
+    "hosts": ParserConfig(Default, seperator=(r"\s")),
 }
 
 
-class ConfigurationFs(VirtualFilesystem):
-    __fstype__: str = "META:registry"
+def parse_config(
+    entry: FilesystemEntry,
+    hint: Optional[str] = None,
+    contents: Optional[ParserContents] = None,
+) -> ConfigParser:
+    parser_type = _select_parser(entry, hint)
+
+    parser = parser_type.create_parser(contents)
+
+    with entry.open() as fh:
+        open_file = io.TextIOWrapper(fh, encoding="utf-8")
+        parser.read_file(open_file)
+
+    return parser
+
+
+def _select_parser(entry: FilesystemEntry, hint: Optional[str] = None) -> ParserConfig:
+    if hint and (parser_type := CONFIG_MAP.get(hint)):
+        return parser_type
 
-    def __init__(self, target: Target, **kwargs):
+    extension = entry.path.rsplit(".", 1)[-1]
+
+    known_extension = CONFIG_MAP.get(extension, ParserConfig(Default))
+    return KNOWN_FILES.get(entry.name, known_extension)
+
+
+def create_entry(
+    fs, path: str, entry, hint: str, collapse: str, seperator: tuple[str], comment_prefixes: tuple[str]
+) -> FilesystemEntry:
+    if entry.is_file():
+        contents = ParserContents(collapse, seperator, comment_prefixes)
+        parser_items = parse_config(entry, hint, contents)
+        return ConfigurationEntry(fs, path, entry, parser_items=parser_items)
+    return entry
+
+
+class ConfigurationTreePlugin(Plugin):
+    __namespace__ = "config_tree"
+
+    def check_compatible(self) -> None:
+        # This should be able to be retrieved, regardless of OS
+        return None
+
+    @lru_cache(128)
+    def __call__(
+        self,
+        path: str = "/",
+        hint: Optional[str] = None,
+        collapse: Optional[Union[bool, set]] = None,
+        seperator: Optional[tuple[str]] = None,
+        comment_prefixes: Optional[tuple[str]] = None,
+    ):
+        """Create a configuration entry from a file, or a COnfigurationFilesystem from a directory.
+
+        If a directory is specified in ``path``, please provide the other arguments in the ``get`` call.
```

Comment:
```suggestion
        If a directory is specified in ``path``, the other arguments should be provided in the ``get`` call if needed.
```
?
They are not mandated if you want to use the default (detected) settings per file are they?